pkgs/servers/audiobookshelf/default.nix

```diff
@@ -0,0 +1,120 @@
+{ lib, stdenv, pkgs, fetchFromGitHub, runCommand, buildNpmPackage, nodejs-16_x, tone, ffmpeg-full, util-linux, libwebp }:
+
+let
+  nodejs = nodejs-16_x;
+
+  pname = "audiobookshelf";
+  version = "2.2.12";
+
+  src = fetchFromGitHub {
+    owner = "advplyr";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-gQuUYS84U48TA8PsNsvyHmWYynmje9umGvCbNXeQrjo=";
+  };
+
+  client = buildNpmPackage {
+    pname = "${pname}-client";
+    inherit version;
+
+    src = runCommand "cd" {} ''
+      cp -r ${src}/client $out
+    '';
+
+    NODE_OPTIONS = "--openssl-legacy-provider";
+
+    npmBuildScript = "generate";
+    npmDepsHash = "sha256-UFmdTWxkOGO73g45NzB51/d1oGOhfNZtOlKSfbTsxPY=";
+  };
+
+  nodeDeps = (import ./node-composition.nix {
+    inherit pkgs nodejs;
+    inherit (stdenv.hostPlatform) system;
+  }).nodeDependencies;
+
+in stdenv.mkDerivation rec {
+  inherit pname version src;
+
+  buildInputs = [ nodejs ffmpeg-full tone util-linux ];
```

Comment:
ffmpeg-full and tone are only used in the wrapper, so putting them here will be a noop.